babel-plugins/add-module-export/src/index.js

```diff
@@ -0,0 +1,23 @@
+"use strict";
+
+module.exports = function({ types: t }) {
+  return {
+    visitor: {
+      Program(path, state) {
+        const id = path.scope.generateUidIdentifier('uid')
+        const constNumDecl = t.variableDeclaration('const', [
```

Comment:
@jridgewell is this the right way to add `export someModuleUniqueId = 42;`? would appreciate advice

Reply:
Almost. Drop the `t.exportSpecifier`:

```js
export default function (babel) {
  const { types: t } = babel;
  
  return {
    name: "ast-transform", // not required
    visitor: {
      Program(path) {
        const uid = path.scope.generateUidIdentifier('uid');
        const declar = t.variableDeclaration('const', [
          t.variableDeclarator(uid, t.numericLiteral(42)),
        ]);
        
        const ex = t.exportNamedDeclaration(declar, []);
        path.unshiftContainer("body", ex);
      }
    }
  };
}
```

http://astexplorer.net/#/gist/9d4d2830b0b2b476627117b6a7798929/e97ac0bf49dd086154e1fd3c3657d90bf330d55c